### PR TITLE
Introduce cleanup logic in helm test

### PR DESCRIPTION
**Pull request by pdostal**
1) Implement k8s old namespace cleanup
2) Assert classic namespace deletion

- Related ticket: [poo#123511](https://progress.opensuse.org/issues/123511)
- Verification run: [pdostal-server.suse.cz/t4000](https://pdostal-server.suse.cz/t4000)

### CODE REVIEW

File: tests/containers/helm.pm [hunk 1]
```diff
@@ -109,6 +112,7 @@ sub run {
     assert_script_run("helm search repo apache");
     assert_script_run("helm show all $chart");
 
+    helm_cleanup() unless $self->{is_k3s};
```
Comment (asmorodskyi) [inline tests/containers/helm.pm:115]: does it make sense that one line below there is variable  with same name but not from `$self` ?

File: tests/containers/helm.pm [hunk 2]
```diff
@@ -121,9 +125,31 @@ sub post_fail_hook {
     script_run("kubectl config set-context --current --namespace=default");
     script_run("kubectl delete namespace helm-ns-$job_id");
 
+    helm_cleanup() unless $self->{is_k3s};
     uninstall_k3s() if $self->{is_k3s};
 }
 
+sub helm_cleanup {
+    diag("helm_cleanup started.");
```
Comment (asmorodskyi) [inline tests/containers/helm.pm:133]: AFAIK diag will only print something in os-autoinst log , maybe better to do record_info instead ? ( or in addition ? )

File: tests/containers/helm.pm [hunk 3]
```diff
@@ -121,9 +125,31 @@ sub post_fail_hook {
...
+    my $namespaces_output = script_output("kubectl get namespaces -o=json");
+    my $namespaces_json = decode_json($namespaces_output);
+    my @namespaces = @{$namespaces_json->{items}};
+    foreach my $item (@namespaces) {
+        next if ($item->{metadata}->{name} !~ /^helm-ns-/);
+        if ($item->{metadata}->{creationTimestamp}) {
+            my $age = time() - str2time($item->{metadata}->{creationTimestamp});
+            if ($age > 10800) { # 3 hours
+                diag("The namespace $item->{metadata}->{name} is $age seconds old. It will be removed.");
+                script_run("kubectl delete all --all -n " . $item->{metadata}->{name});
```
Comment (asmorodskyi) [inline tests/containers/helm.pm:143]: I am not sure that it is good idea to put something like into the test , imagine that 2/3/5 tests will execute such code in parallel we can catch some not nice bugs in k8s cluster ...

File: tests/containers/helm.pm (hunk 2 above)
Comment (pdostal) [reply tests/containers/helm.pm:133]: I switched from `record_info` to `diag` to not clutter the test overview that much. those messages are anyways not necessary for the 'test' - they're just 'administrative' informations.

File: tests/containers/helm.pm (hunk 1 above)
Comment (pdostal) [reply tests/containers/helm.pm:115]: Good catch

File: tests/containers/helm.pm (hunk 3 above)
Comment (asmorodskyi) [reply tests/containers/helm.pm:142]: > I don't understand. Only 3 hours old namespaces will be deleted. So most of the time none. If none is deleted then the only API call this does is `kubectl get namespaces -o=json` which is pretty minimal.

containers tests are running in really different setup for different versions of SLE , please tell me how many times this code will run during one day ? 

> But there is question if this should be here or in PCW...

IMO this task is 100% match PCW and 0% match test code

Comment (ilausuch) [reply tests/containers/helm.pm:142]: I agree with Anton, this should be moved to PCW. We had similar discussion for jobs and the conclussions were that the PCW is the best place to do this

**Pull request closed without merging**